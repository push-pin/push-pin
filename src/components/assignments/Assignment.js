import React from 'react';
import PropTypes from 'prop-types';

function Assignment({ assignment }) {
  let gradeDisplay = 'Incomplete';
  if(assignment.grade) {
    gradeDisplay = 'Completed';
  }

  return (
    <>
      <p>{assignment.classDate}</p>
      <p>{assignment.title}</p>
      <p>Submitted: {assignment.submitted}</p>
      <p>{gradeDisplay}</p>
    </>
  );
}

Assignment.propTypes = {
  assignment: PropTypes.shape({
    classDate: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    submitted: PropTypes.bool.isRequired,
    // will eventually be a number if graded
    grade: PropTypes.bool.isRequired
  }).isRequired
};

export default Assignment;
