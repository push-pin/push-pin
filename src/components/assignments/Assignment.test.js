import React from 'react';
import { shallow } from 'enzyme';
import Assignment from './Assignment';

describe('Assignment component', () => {
  it('renders a Assignment', () => {
    const assignment = {
      date: new Date(),
      title:  'Ass 1',
      grade: false,
      submitted: false
    };
    const wrapper = shallow(<Assignment assignment={assignment} />);
    expect(wrapper).toMatchSnapshot();
  });
});
