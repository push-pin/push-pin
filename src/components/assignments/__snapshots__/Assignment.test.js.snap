// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Assignment component renders a Assignment 1`] = `
<assContainer>
  <p>
    <Component />
  </p>
  <p>
    Ass 1
  </p>
  <p>
    Submitted: 
  </p>
  <p>
    Incomplete
  </p>
</assContainer>
`;
